hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -1408,8 +1408,7 @@ aggregate st = \case
             { coordinatedHeadState =
                 coordinatedHeadState
                   { localUTxO = newLocalUTxO
-                  , -- NOTE: union is left biased, does it matter to us here?
-                    pendingDeposits = pendingDeposits `Map.union` existingDeposits
+                  , pendingDeposits = Map.toList . Map.fromList $ pendingDeposits <> existingDeposits
```

Comment:
It does compile but I wanted to make sure there are no duplicate keys in this list

Reply:
So the union would also not produce duplicates (and is left biased while doing that). We don't really expect duplicates here since the TxId's are reasonably unique anyway but list type doesn't give us the same guarantees as the Map does so I thought it might be good to be really sure we are not appending two same commits.